pkg/k8s/apis/cilium.io/v2alpha1/bgp_peer_types.go

```diff
@@ -0,0 +1,148 @@
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumBGPPeerConfigList is a list of CiliumBGPPeer objects.
+type CiliumBGPPeerConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumBGPPeer.
+	Items []CiliumBGPPeerConfig `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium,ciliumbgp},singular="ciliumbgppeerconfig",path="ciliumbgppeerconfigs",scope="Cluster",shortName={bgppeer}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+// +kubebuilder:storageversion
+
+type CiliumBGPPeerConfig struct {
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec is the specification of the desired behavior of the CiliumBGPPeerConfig.
+	Spec CiliumBGPPeerConfigSpec `json:"spec,omitempty"`
+}
+
+type CiliumBGPPeerConfigSpec struct {
+	// Transport defines the BGP transport parameters for the peer.
+	//
+	// +kubebuilder:validation:Optional
+	Transport *CiliumBGPTransport `json:"transport,omitempty"`
+
+	// Timers defines the BGP timers for the peer.
+	//
+	// +kubebuilder:validation:Optional
+	Timers *CiliumBGPTimers `json:"timers,omitempty"`
+
+	// GracefulRestart defines graceful restart parameters which are negotiated
+	// with this peer. If empty / nil, the graceful restart capability is disabled.
+	//
+	// +kubebuilder:validation:Optional
+	GracefulRestart *CiliumBGPNeighborGracefulRestart `json:"gracefulRestart,omitempty"`
+
+	// EBGPMultihopTTL controls the multi-hop feature for eBGP peers.
+	// Its value defines the Time To Live (TTL) value used in BGP
+	// packets sent to the peer.
+	// The value 1 implies that eBGP multi-hop feature is disabled
+	// (only a single hop is allowed).
+	//
+	// This field is ignored for iBGP peers.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=255
+	// +kubebuilder:default=1
+	EBGPMultihop *int32 `json:"ebgpMultihop,omitempty"`
+
+	// Families, if provided, defines a set of AFI/SAFIs the speaker will
+	// negotiate with it's peer.
+	//
+	// If this slice is not provided the default families of IPv6 and IPv4
+	// will be provided.
+	//
+	// +kubebuilder:validation:Optional
+	Families []CiliumBGPFamily `json:"families,omitempty"`
+}
+
+// CiliumBGPFamily represents a AFI/SAFI address family pair.
+type CiliumBGPFamily struct {
+	// Afi is the Address Family Identifier (AFI) of the family.
+	//
+	// +kubebuilder:validation:Enum=ipv4;ipv6;l2vpn;ls;opaque
```

Comment:
I know this is a carryover but enums should be camel case: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants. If you agree, update throughout the new APIs.

Reply:
This is a common structure used between old and new API, I do not want to modify it as it will break backward compatibility.

Reply:
One option is to create a new struct that is identical to CiliumBGPFamily but with camel-cased enums and is used by the new CRD. Not a bid deal, just a thought to align new CRDs with k8s API conventions as best as possible. Feel free to resolve this comment if you prefer sticking with the current enums.

Reply:
I am inclined to leave it as is. I understand camel case is standard practice, however, SAFI enums are getting a bit funky ( all the network acronyms when capitalized and underscore removed became unreadable ). 

Current mapping seems much more user-friendly.

Reply:
OK, thanks for the feedback. You can consider this comment resolved.